### PR TITLE
Improved Skopt NeptuneMonitor

### PR DESCRIPTION
**Pull request by jakubczakon**
Added:
* logging checkpoints of results object every epoch
* logging regret plot after training is finished

Changed
* Renamed `NeptuneMonitor` to `NeptuneCallback`
* Dropped functions that log single objects and just left the `log_results` function

### CODE REVIEW

File: neptunecontrib/monitoring/skopt.py [hunk 1]
```diff
@@ -81,246 +95,95 @@ def log_results(results, experiment=None):
...
-
-     """
-    log_best_score(results, experiment)
-    log_best_parameters(results, experiment)
-    log_plot_convergence(results, experiment, channel_name='diagnostics')
-    log_plot_evaluations(results, experiment, channel_name='diagnostics')
-    log_plot_objective(results, experiment, channel_name='diagnostics')
-
-
-def log_runs(results, experiment=None):
```
Comment (pitercl) [inline neptunecontrib/monitoring/skopt.py:101]: Do you feel we don't need to be backward compatible here?

File: neptunecontrib/monitoring/skopt.py [hunk 2]
```diff
@@ -62,7 +74,7 @@ def _get_last_params(res):
         return str(named_params)
 
 
-def log_results(results, experiment=None):
+def log_results(results, experiment=None, log_plots=True, log_pickle=True):
     """Logs runs results and parameters to neptune.
 
     Logs all hyperparameter optimization results to Neptune. Those include best score ('best_score' channel),
```
Comment (pitercl) [inline neptunecontrib/monitoring/skopt.py:80]: The `diagnostics` channel doesn't exist in this implementation any more.
Also, we call them "logs" these days :)

File: neptunecontrib/monitoring/skopt.py [hunk 3]
```diff
@@ -81,246 +95,95 @@ def log_results(results, experiment=None):
...
+        fig = axes2fig(sk_plots.plot_objective(results), fig=fig)
+        experiment.log_image('plot_objective', fig)
+    except Exception as e:
+        print('Could not create the objective chart due to error: {}'.format(e))
 
-        Send skopt plot_objective figure to neptune::
 
-            import neptune
-            import neptunecontrib.monitoring.skopt as sk_utils
+def _log_results_object(results, experiment=None):
```
Comment (pitercl) [inline neptunecontrib/monitoring/skopt.py:173]: I'd feel a lot safer if `experiment` didn't default to `None`, since it's called in the next line :)

Comment (pitercl) [reply neptunecontrib/monitoring/skopt.py:173]: Actually, this goes for all the helpers in this file.

File: neptunecontrib/monitoring/skopt.py [hunk 4]
```diff
@@ -81,246 +95,95 @@ def log_results(results, experiment=None):
...
-        fig = axes2fig(sk_plots.plot_objective(results), fig=fig)
-        with tempfile.NamedTemporaryFile(suffix='.png') as f:
-            fig.savefig(f.name)
-            _exp.send_image(channel_name, f.name)
-    except Exception as e:
-        print('Could not create ans objective chart due to error: {}'.format(e))
+    return buffer
 
 
 def _format_to_named_params(params, result):
```
Comment (pitercl) [inline neptunecontrib/monitoring/skopt.py:189]: The parentheses in this function are not needed.

File: neptunecontrib/monitoring/skopt.py (hunk 1 above)
Comment (jakubczakon) [reply neptunecontrib/monitoring/skopt.py:101]: I really don't think those functions were used separately. 
I can add deprecation warnings but I feel this is not necessary.

File: neptunecontrib/monitoring/skopt.py (hunk 2 above)
Comment (jakubczakon) [reply neptunecontrib/monitoring/skopt.py:80]: Mhm, will revert to that.

File: neptunecontrib/monitoring/skopt.py (hunk 3 above)
Comment (jakubczakon) [reply neptunecontrib/monitoring/skopt.py:173]: Yep, dropped when refactoring

File: neptunecontrib/monitoring/skopt.py (hunk 4 above)
Comment (jakubczakon) [reply neptunecontrib/monitoring/skopt.py:189]: Done

File: neptunecontrib/monitoring/skopt.py (hunk 3 above)
Comment (jakubczakon) [reply neptunecontrib/monitoring/skopt.py:173]: Done

File: neptunecontrib/monitoring/skopt.py (hunk 2 above)
Comment (jakubczakon) [reply neptunecontrib/monitoring/skopt.py:80]: Done

File: neptunecontrib/monitoring/skopt.py (hunk 1 above)
Comment (pitercl) [reply neptunecontrib/monitoring/skopt.py:101]: Ack, I'm deferring to you on that.

**Pull request merged**